### PR TITLE
[batch] Add query parameter for listing jobs

### PR DESCRIPTION
**Pull request by jigold**
FYI @konradjk

### CODE REVIEW

File: batch/test/test_batch.py [hunk 1]
```diff
@@ -203,6 +203,30 @@ def assert_batch_ids(expected, q=None):
 
         assert_batch_ids({b2.id}, f'tag={tag} name=b2')
 
+    def test_list_jobs(self):
+        b = self.client.create_batch()
+        j_success = b.create_job('ubuntu:18.04', ['true'])
+        j_failure = b.create_job('ubuntu:18.04', ['false'])
+        j_error = b.create_job('ubuntu:18.04', ['sleep 5'], attributes={'tag': 'bar'})
+        j_running = b.create_job('ubuntu:18.04', ['sleep', '1800'], attributes={'tag': 'foo'})
```
Comment (danking) [inline batch/test/test_batch.py]: hmm. when I did something like this cotton said that I had to make the job actually run forever, so I [added a while true loop](https://github.com/hail-is/hail/blob/02072feee4d5933261c8506b5cdd3990b56c8e59/batch/test/test_dag.py#L68-L71). I see that this file frequently uses long sleeps in place of an actual run forever program. I guess that's fine

File: batch/test/test_batch.py [hunk 2]
```diff
@@ -203,6 +203,30 @@ def assert_batch_ids(expected, q=None):
...
+
+        def assert_job_ids(expected, q=None):
+            actual = set([j['job_id'] for j in b.jobs(q=q)])
+            assert actual == expected
+
+        assert_job_ids({j_success.job_id}, 'success')
+        assert_job_ids({j_success.job_id, j_failure.job_id, j_error.job_id}, 'done')
+        assert_job_ids({j_running.job_id}, '!done')
+        assert_job_ids({j_running.job_id}, 'tag=foo')
+        assert_job_ids({j_error.job_id, j_running.job_id}, 'has:tag')
```
Comment (danking) [inline batch/test/test_batch.py]: add a test for `q=None`.
